README.md

```diff
@@ -312,70 +344,96 @@ const commonProps = {
+Although imgix is open to feature suggestions, we might not accept the feature if it is a very specific use case. The features below are examples of what we consider general advanced use cases. Our target here is to support 95% of all the usages of normal `img`, `picture`, and `source` elements.
+
+If your desired feature falls outside this percentage, do not worry! You will probably still be able to achieve your feature with react-imgix's more powerful api: `buildUrl`.
+
+This library exposes a pure function, `buildURL`, for generating full imgix urls given a base url and some parameters.
+
+```jsx
+import { buildURL } from "react-imgix";
+
+buildURL("http://yourdomain.imgix.net/image.png", { w: 450, h: 100 }); // => http://yourdomain.imgix.net/image.png?auto=format&w=450&h=100&ixlib=react-x.x.x
 ```
 
-This works for Source and Picture elements as well.
+The base url may also contain query parameters. These will be overriden by any parameters passed in with the second parameter.
 
-#### Background mode
+This feature can be used to create your own custom `img` elements, or for use with other image components, such as [React-bootstrap's Image component](https://react-bootstrap.github.io/components/images/).
 
-Images can be rendered as a background behind children by using `<Background />`. The component will measure the natural size of the container as determined by the CSS on the page, and will render an optimal image for those dimensions.
+The `ixlib` parameter may be disabled by: `buildURL(<url>, <params>, { disableLibraryParam: true })`
 
-Example:
+#### Passing custom html attributes
 
-```jsx
-// In CSS
-.blog-title {
-  width: 100vw;
-  height: calc(100vw - 100px);
-}
+This library allows the developer to pass any attribute they like to the underlying DOM element with `htmlAttributes`.
 
-// In Component (React)
-import { Background } from 'react-imgix'
+For example, if the the developer would like to attach a custom onLoad callback to an img component:
 
-<Background src="https://.../image.png" className="blog-title">
-  <h2>Blog Title</h2>
-</Background>
+```jsx
+<Imgix
+  src="..."
+  sizes="..."
+  htmlAttributes={{
+    onLoad: () => handleImgOnLoad,
+  }}
+/>
 ```
 
-This component shares a lot of props that are used in the main component, such as `imgixParams`, and `htmlAttributes`.
+#### Lazy Loading
 
-As the component has to measure the element in the DOM, it will mount it first and then re-render with an image as the background image. Thus, this technique doesn't work very well with server rendering. If you'd like for this to work well with server rendering, you'll have to set a width and height manually.
+If you'd like to lazy load images, we recommend using [lazysizes](https://github.com/aFarkas/lazysizes). In order to use react-imgix with lazysizes, you can simply tell it to generate lazysizes-compatible attributes instead of the standard `src`, `srcset`, and `sizes` by changing some configuration settings:
```

Comment:
Not something this PR should be concerned with, but something we should consider for the future: should we still recommend Lazysizes? The new [`loading` attribute](https://caniuse.com/#search=lazy-load) is available for something like half of internet users right now, it's becoming increasingly weird that we don't even mention it in this section (and in similar sections on a bunch of our other READMEs).

Reply:
I agree we should add this to the README. I think it's out of scope of this PR, and weak opinion: I think it makes more sense to add it to all repositories at the same time.